Modernize scaffold generator

dhh opened the pull request:
The default scaffold generator should be encouraging the use of partials. It also shouldn't try so hard to style things. We can do such styling as a build-on-top via CSS framework gems, like tailwindcss-rails.

Co-authored-by: Abhay Nikam <nikam.abhay1@gmail.com>

shioyama committed:
Don't bother checking if strings respond to string methods

The respond_to? calls here are checking if the unsafe method names, in
this case "gsub" and "sub", respond to those same methods. This is
nonsensical and unnecessary.

alpaca-tc committed:
`connected_to` shouldn't be called on the abstract class that not established connection

Fixed: https://github.com/rails/rails/issues/40559#issuecomment-752056106

When abstract class hasn't own connections, calling `AbstractClass.connection`
returns parent class's connection. We call `AbstractClass.connection.preventing_writes?`
expecting abstract class's state to be returned, but actually it is parent's one.

I think that it isn't expected behavior so I prevents call `connected_to` on the abstract
class that not established the connection.

etiennebarrie committed:
Ensure test rake commands run immediately

Before this commit, Rails test Rake tasks only load the test files, and
the tests only run in an at_exit hook via minitest/autorun.

This prevents conditionally running tasks only when tests pass, or even
more simply in the right order. As a simple example, if you have:

task default: [:test, :rubocop]

The rubocop task will run after the test task loads the test files but
before the tests actually run.

This commit changes the test Rake tasks to shell out to the test runner
as a new process.

This diverges from previous behavior because now, any changes made in
the Rakefile or other code loaded by Rake won't be available to the
child process. However this brings the behavior of `rake test` closer to
the behavior of `rails test`.

Co-authored-by: Adrianna Chang <adrianna.chang@shopify.com>

radar committed:
Use major + minor AR versions in 'Directly inheriting' error message

gmcgibbon committed:
Correct logic warning about initializers

Defining initializer 4 to run before itself and after 3 is possible, but
not when the before constraint contradicts the after constraint via 2.

malomalo committed:
Connection specification now passes the "url" key to the adapter

If the "url" protocol is "jdbc", "http", or "https" the url option will
be passed to the adapter. Previously only urls with the "jdbc" prefix
were passed to the Active Record Adapter, others are assumed to be
adapter specification urls.

Fixes #41137.

hummusonrails committed:
switch references to main branch in docs

MikeRogers0 committed:
Updating references to /rails/blob/master & raiks/tree/master to point to main

ghiculescu committed:
Add Webpacker to Guides list

Adds a link to the Webpacker guide (added in https://github.com/rails/rails/pull/40817) to the [guides list](https://guides.rubyonrails.org/) and "Guides Index" dropdown menu.

guillaumebriday committed:
Adding badges and logo to README and CONTRIBUTING page

ghiculescu committed:
Webpacker guide: remove Basecamp reference [docs]

As a Rails user, if you *don't* know who Basecamp is or their relationship to the Rails framework, then this sentence is confusing. Reworded to instead just refer to Rails defaults.

cc @rossta

ghiculescu committed:
fix broken link

ghiculescu committed:
Update webpacker.md

davidstosik committed:
Add small improvements to the Webpacker guide

- fix typos like double spaces, accidental caps
- improve some HTML links by giving them a label
- etc

kamipo committed:
Raise unknown type error on the definition time

If unknow type is given for attribute (`attribute :foo, :unknown`),
unknown type error isn't raised on the definition time but runtime.

It should be raised on the definition time.

eileencodes committed:
Fix granular connection swapping when there are multiple abstract classes

Some applications maybe have multiple abstract classes in the
inheritance chain but only one of those abstract classes is the one we
want to switch connections on. Previously, multiple abstract class
inhertance would break `connected_to` and not switch models to the
correct connection context.

To fix this we added a boolean that is set on the class when a
connection is established so we can check for whether it's identified as
a `connection_class?`. This allows us to delete the `abstract_class?`
check, since you cannot establish a connection on a non-abstract class.

The existing tests were changed because they were not calling
`connects_to` and granular swapping won't work on classes that didn't
establish the connection. The issue in these tests will be prevented
when #40965 is merged.

Co-authored-by: John Crepezzi <john.crepezzi@gmail.com>

santib committed:
ActionCable guides suggest test adapter for test env [ci skip]

jhawthorn committed:
Use length == 0 instead of empty in Preloader

Previously we were checking empty? on the association which would make a
query. Instead we can check length == 0 to ensure we are using the
length of the loaded records and not issuing extra queries.

Co-authored-by: Dinah Shi <dinahshi@github.com>

rafaelfranca committed:
Rename master to main in all code references

jhawthorn committed:
Add regression tests for preloader query count

MikeRogers0 committed:
Changing 'rails new' --master to be --main

Renaming test containing flag

Updating other test referencing master branch

Add notice that --master is deprecated, but still working the same as --main

Only set @main if it's nil

Making warn wildcard

I think a hidden aliaes would be just as good

Improving description & fixing rubocop error

Forgot comma

Deprecation warning was kind of hard - so just doing alias for now

rubocop -a

ayrton committed:
Allow to opt-out of `strict_loading` mode on a per-record base.

This is useful when strict loading is enabled application wide or on a
model level.

MikeRogers0 committed:
I think passing in the --master argument to run_generator is the way to go

MikeRogers0 committed:
Removing .count to figure out why its failing

ghiculescu committed:
Update test docs in "contributing to Rails" guide

Updated guide to recommend `bin/test` and show examples of its use in different contexts.

Co-authored-by: Marivaldo Cavalheiro <marivaldo@gmail.com>

kamipo committed:
Refactor `attribute` not to reference `&block`

intrip committed:
Raise error when passing passing a class to :source_type

Raise `ArgumentError` when `:source_type` is given as class
in `has_many :through` relation.

Closes #41092

ijlee2 committed:
Added the unless-block for continuity

ijlee2 committed:
Added :status for continuity

ijlee2 committed:
Removed line for continuity

kamipo committed:
Don't return query cache enabled pools in the query cache executor

Follow up #41046.

If query cache is enabled for all connection pools, `pools` always
matches to `all_connection_pools`, returning `pools` has become
redundant.

Commit:
Remove trailing commas in advanced route constraints example

jonathanhefner committed:
Run bin/yarn via Ruby

Windows cannot directly run shebang scripts, such as `bin/yarn`.
Therefore, run `bin/yarn` via Ruby.

Actually fixes #40942.
Fixes #41123.

jonathanhefner committed:
Search for yarn.cmd and yarn.ps1 in bin/yarn

Follow-up to #40950.

On Windows, Yarn actually uses `yarn.cmd` and `yarn.ps1` for PowerShell.

jonathanhefner committed:
Avoid testing Thor internals

Thor's `apply` method is responsible for fetching a template when given
a URL.  Therefore, assume that `apply` behaves correctly, and simply
test that `apply` is called correctly.

This avoids errors like https://buildkite.com/rails/rails/builds/74245#540ecdf1-58ea-470a-a397-09f675520eb9/1100-1109
resulting from erikhuda/thor@4ce38c5478a5a13f2214ca833eafc167615b4e6a.

rafaelfranca committed:
Fix the benchmark script to point to the main branch

jonathanhefner committed:
Resolve default annotation tags after config loads

`Rails::SourceAnnotationExtractor::Annotation.tags` may be modified by
app configuration.  Therefore, resolve default annotation tags after
loading the app configuration.

This fixes errors like https://buildkite.com/rails/rails/builds/74268#240d60bc-baa7-4b6e-ad21-b3172095f939/1083-1092
resulting from erikhuda/thor@0222fe52ed3803fe3ee0033da5b6faac5ee6299c.

rafaelfranca committed:
Upgrade all the gems to make sure we are testing with the latest versions locally

intrip committed:
Fix current_page? with kwargs on ruby3

Prevent raising an error when `options` are given as
kwargs, this is done by overriding `options` with kwargs
if `options` are `nil`; this implies that if both `options` and
kwargs are given, `options` takes precedence.

Fixes #41198

Aguardientico committed:
Support hash config for `structure_dump_flags` and `structure_load_flags` flags

    Now that Active Record supports multiple databases configuration
    we need a way to pass specific flags for dump/load databases since
    the options are not the same for different adapters.
    We can use in the original way:
    ```ruby
    ActiveRecord::Tasks::DatabaseTasks.structure_dump_flags = ['--no-defaults', '--skip-add-drop-table']
    #or
    ActiveRecord::Tasks::DatabaseTasks.structure_dump_flags = '--no-defaults --skip-add-drop-table'
    ```
    And also use it passing a hash, with one or more keys, where the key
    is the adapter
    ```ruby
    ActiveRecord::Tasks::DatabaseTasks.structure_dump_flags = {
      mysql2: ['--no-defaults', '--skip-add-drop-table'],
      postgres: '--no-tablespaces'
    }
    ```

ttilberg committed:
Update name of input to fix typo

ijlee2 committed:
Added a new line after the include statement

Commit:
Slim down scaffold css

To prevent conflicts with utility frameworks that might also be resetting base elements.

Commit:
Use a box-style partial rather than a table

Shows the usage of partials right from the start. Better compatibility with upgrading to Turbo frames/stream updates.

Commit:
Correct use of quotes

Commit:
Use modern array-of-strings declaration

Commit:
Use double quotes for everything

fxn committed:
Update documents.yaml

Commit:
Fix syntax

Commit:
Remove outdated viewport declaration

This should be handled in app stylesheets.

dhh commented:
If anyone would like to help me wrap this up with correcting tests, I'd appreciate it!

Commit:
Use double quotes everywhere

Diogomartf commented:
> If anyone would like to help me wrap this up with correcting tests, I'd appreciate it!

I would like to help, but how do I know which tests I need to correct? All that are failing?  
There are some that don't seem related.

Sorry if that might be obvious but it's my first time here ✌️

dhh commented:
Thanks @Diogomartf! Yes, basically get this PR passing. There are some failing tests in the general suite that won't stop the overall build from passing. You can ignore that. Just all the ones focused on these changes ✌️❤️

rafaelfranca commented on railties/lib/rails/generators/erb/scaffold/templates/new.html.erb.tt:
```diff
@@ -1,5 +1,9 @@
-<h1>New <%= singular_table_name.titleize %></h1>
+<h1>New <%= singular_table_name %></h1>
```
Why was titleize removed from here? Do we want to generate `New user_account`?

dhh replied:
Ahh, I was just looking to get rid of the capitalization. I guess we'd need to do #human_name instead?

rafaelfranca commented on railties/lib/rails/generators/rails/scaffold_controller/templates/api_controller.rb.tt:
```diff
@@ -4,7 +4,7 @@ require_dependency "<%= namespaced_path %>/application_controller"
 <% module_namespacing do -%>
 class <%= controller_class_name %>Controller < ApplicationController
-  before_action :set_<%= singular_table_name %>, only: [:show, :update, :destroy]
+  before_action :set_<%= singular_table_name %>, only: %w[ show update destroy ]
```
should not it be `%i`?

eileencodes committed:
Fix unintialized instance variable connection_class

This was throwing a warning when the test suite is run. Adding an
reader fixes the issue.

etiennebarrie committed:
Allow jobs to rescue all exceptions

Before this commit, only StandardError exceptions can be handled by
rescue_from handlers.

This changes the rescue clause to catch all Exception objects, allowing
rescue handlers to be defined for Exception classes not inheriting from
StandardError.

This means that rescue handlers that are rescuing Exceptions outside of
StandardError exceptions may rescue exceptions that were not being
rescued before this change.

Co-authored-by: Adrianna Chang <adrianna.chang@shopify.com>

andrehjr committed:
Remove legacy media=screen default from stylesheet_link_tag.

If the media attribute is omitted, the default for web browsers is "all", meaning that by default links apply to all media.

Before:

```ruby
> stylesheet_link_tag "style"
=> <link href="/assets/style.css" media="screen" rel="stylesheet" />
```

After:

```ruby
> stylesheet_link_tag "style"
=> <link href="/assets/style.css" rel="stylesheet" />

```

The current behavior is not going to change for existing applications.

For newly built applications, the media attribute is not going to be added by default. Which can be configured using the following:

```
Rails.application.config.action_view.stylesheet_media_default = false
```

kamipo committed:
`start_with?` allows multiple prefix values

andrehjr committed:
Fix doc: stylesheet_include_tag -> stylesheet_link_tag

janko committed:
Handle throwing in controller action in log subscriber

When throw was used in a controller action, and there is matching catch
around the request in a Rack middleware, then :exception won't be
present in the event payload.

This is because ActiveSupport::Notifications::Instrumenter.instrument
sets :exception in a rescue handler, but rescue is never called in a
throw/catch scenario:

  catch(:halt) do
    begin
      throw :halt
    rescue Exception => e
      puts "rescue" # never reached
    ensure
      puts "ensure"
    end
  end

Missing :exception was actually handled prior to Rails 6.1.0, but an
optimization updated the code to assume this was present. So this can be
considered a regression fix.

houhoulis committed:
Fix code block in Webpacker Guide [ci skip]

Formatting fix: the example directory tree listing was bleeding into the
previous paragraph and had extraneous characters due to a missing
newline.

kamipo committed:
Remove wrong usage for `arel_table` [ci skip]

This usage doesn't work properly (missing `where`, undefined `published`,
`arel_table` in the scope definition accidentally lose table alias).

composerinteralia committed:
Restore ActiveStorage::Blob#find_signed

Rails 6.0 had a [public `find_signed` method][docs], but we changed it
to `find_signed!` in https://github.com/rails/rails/commit/31148cd6bef4f3a3059c51c587a8bff78a2e73e3.

This commit adds back `find_signed` alongside `find_signed!` to match
the corresponding [Active Record methods][].

[docs]: https://api.rubyonrails.org/v6.0.0/classes/ActiveStorage/Blob.html#method-c-find_signed
[Active Record methods]: https://github.com/rails/rails/blob/main/activerecord/lib/active_record/signed_id.rb#L42-L66

Commit:
Use symbols not strings for before_action scoping

Commit:
Use human name to deal with double word records

Commit:
Grab test fixes from #41219

Thanks @abhaynikam 🙏

seanpdoyle committed:
Improve Fixture support for Active Storage (#41065)

* Improve Fixture support for Active Storage

Inspired by [76b33aa][], this commit extends the Active Storage
documentation to elaborate on how to declare fixtures.

In support of that, also introduce the `ActiveStorage::FixtureSet.blob`
method for injecting in-line [ActiveStorage::Blob][] attributes directly
into fixture YAML.

[76b33aa]: https://github.com/rails/rails/commit/76b33aa3d102ad4ff3f766ac2d3a711d73aaee82
[ActiveStorage::Blob]: https://edgeapi.rubyonrails.org/classes/ActiveStorage/Blob.html

* Extra CR for style

* Two-space indention

* Explaining variable didn't explain, inline for style

Co-authored-by: David Heinemeier Hansson <david@loudthinking.com>

abhaynikam commented on railties/lib/rails/generators/erb/scaffold/templates/show.html.erb.tt:
```diff
@@ -1,19 +1,10 @@
+  <%%= link_to "Edit this <%= human_name.downcase %>", edit_<%= singular_table_name %>_path(@<%= singular_table_name %>) %> |
+  <%%= link_to "Back to <%= human_name.pluralize.downcase %>", <%= index_helper %>_path %>
+
+  <%%= button_to "Destroy this <%= human_name.downcase %>", <%= singular_table_name %>_path(@<%= singular_table_name %>), method: :delete %>
```
Maybe `link_to` would be better? 

```suggestion
  <%%=link_to "Destroy this <%= human_name.downcase %>", <%= singular_table_name %>_path(@<%= singular_table_name %>), method: :delete, data: { confirm: 'Are you sure?' } %>
```

dhh replied:
We are looking to phase out links that perform DELETE. It's not a good, safe pattern.

seanpdoyle committed:
Improve ActionText::FixtureSet documentation (#41062)

* Improve ActionText::FixtureSet documentation

Support for Action Text attachments in fixtures was added by [76b33aa][] and
released as part of [6.1.1][], but has not yet been documented.

This commit documents the `ActionText::FixtureSet` for the API
documentation, and mentions it in the Rails Guides pages.

[76b33aa]: https://github.com/rails/rails/commit/76b33aa3d102ad4ff3f766ac2d3a711d73aaee82
[6.1.1]: https://github.com/rails/rails/releases/tag/v6.1.1

* Fix indention of comments

Co-authored-by: David Heinemeier Hansson <david@loudthinking.com>

Commit:
Fix tests

Commit:
Use locar var not ivar

Commit:
Fix capitalization change

hahmed commented on railties/lib/rails/generators/rails/app/templates/app/views/layouts/application.html.erb.tt:
```diff
@@ -2,19 +2,18 @@
-    <%%= stylesheet_link_tag 'application', media: 'all' %>
-    <%%= javascript_pack_tag 'application' %>
+    <%%= stylesheet_link_tag "application" %>
+    <%%= javascript_pack_tag "application", media: "all" %>
```
```suggestion
    <%%= javascript_pack_tag "application" %>
```

Looks like we can drop the media: "all" as it's not needed for javascript tags

Commit:
Update railties/lib/rails/generators/rails/app/templates/app/views/layouts/application.html.erb.tt

Co-authored-by: Haroon Ahmed <haroon.ahmed25@gmail.com>

hahmed commented on railties/test/generators/scaffold_controller_generator_test.rb:
```diff
@@ -221,14 +221,11 @@ def test_model_name_option
-      assert_match("'Edit', edit_admin_user_path(user)", content)
-      assert_match("'Destroy', [:admin, user]", content)
-      assert_match("'New User', new_admin_user_path", content)
+      assert_match("'New user', new_admin_user_path", content)
```
```suggestion
      assert_match("\"New user\", new_admin_user_path", content)
```

(double quote fixes the test)

hahmed replied:
test failing:


ScaffoldControllerGeneratorTest#test_model_name_option [test/generators/scaffold_controller_generator_test.rb:223]:
--
  | Expected /'New\ user',\ new_admin_user_path/ to match "<p id=\"notice\"><%= notice %></p>\n\n<h1>User</h1>\n\n<div id=\"users\">\n  <%= render @users %>\n</div>\n\n<%= link_to \"New user\", new_admin_user_path %>\n".

hahmed commented on railties/test/generators/scaffold_controller_generator_test.rb:
```diff
@@ -221,14 +221,11 @@ def test_model_name_option
 
     assert_file "app/views/admin/users/new.html.erb" do |content|
-      assert_match("'Back', admin_users_path", content)
+      assert_match("\"Back to user\", admin_users_path", content)
```
```suggestion
      assert_match("\"Back to users\", admin_users_path", content)
```

the s was missing, in users which fixes this test.

Commit:
Update railties/test/generators/scaffold_controller_generator_test.rb

Co-authored-by: Haroon Ahmed <haroon.ahmed25@gmail.com>

Commit:
Update railties/test/generators/scaffold_controller_generator_test.rb

Co-authored-by: Haroon Ahmed <haroon.ahmed25@gmail.com>

Commit:
Fix test

jonathanhefner commented on railties/test/generators/scaffold_generator_test.rb:
```diff
@@ -282,10 +282,7 @@ def test_scaffold_with_namespace_on_invoke
-      assert_match("'Edit', edit_admin_role_path(admin_role)", content)
-      assert_match("'Destroy', admin_role", content)
-      assert_match("'New Admin Role', new_admin_role_path", content)
+      assert_match("'New role', new_admin_role_path", content)
```
```suggestion
      assert_match(%("New role", new_admin_role_path), content)
```

jonathanhefner commented:
What do you think about extracting the flash message to a partial as well?  That would make it easier to style with utility classes a la Tailwind.  We could implement `FlashHash#to_partial_path` in a follow-up PR so that the render call could be simply `<%= render flash %>`.

t27duck committed:
Guides: Missing erb tags for stylesheet_pack_tag [ci skip]

The section on needing to use `stylesheet_pack_tag` was lacking the erb
%'s.

I updated the markdown to actually show the helper being used with erb
as well as improve the wording of the sentence.

seanpdoyle committed:
Fix Flaky ActiveStorage test (#41225)

Fixes a flaky Active Storage test introduced by [rails/rails#41065][],
and improves the documentation.

It seems that the test is covering the backwards compatibility of an
older interface for retrieving records through
`ActiveStorage::Record#find_signed!`. The test itself would pass
unpredictably. To isolate the failure and reproduce it consistently, a
see value was found after some trial and error:

```
SEED=59729 bin/test test/fixture_set_test.rb test/models/attachment_test.rb
```

This _used_ to pass consistently because [rails/rails][#41065]
introduced a call to `fixtures :all`, which introduces more variation in
the database's ID generation sequence. Without that line, `id` values
start at `1`, so the fact that calls to
`ActiveStorage::Attached::One#id` and `ActiveStorage::Blob#id` **both
return `1`** is purely coincidence.

The proposed resolution changes the test slightly. Prior to this change,
the identifier used during retrieval and verification fetched from
`@user.avatar.id`, where `@user.avatar` is an instance of
`ActiveStorage::Attached::One`. The verifier/retriever combination in
that test expected a signed identifier for an `ActiveStorage::Blob`
instance. The change involved retrieving an instance through
`@user.avatar.blob`.

To better emphasize how global the `fixtures :all` declaration is, move
it from the `test/fixture_set_test.rb` file to the `test/test_helper.rb`
file.

[rails/rails#41065]: https://github.com/rails/rails/pull/41065

seanpdoyle committed:
Move ActiveStorage fixture hooks to on_load

In a test environment, rely on the loading of
`:active_support_test_case`.

Introduce the `:active_record_fixture_set` hook for the Active Storage
engine to listen for during the load process in a development
environment (like when running `db:fixtures:load`).

Since this commit moves the task-aware path resolution out of the block
that provided local variables, it recreates part of the fixture
directory path resolution logic.

yboulkaid committed:
Remove reference to globalize gem

cveneziani committed:
Improve ActiveRecord strict_loading documentation

Apply the same structure as the documentation of `validate` option by
starting the description with "when set to true". This implies, that
the value passed to the option should be a boolean.

orhantoy committed:
Fix typo [ci skip]

Commit:
Remove SET NAMES, set collation using variable

Fixes malformed packet error that occurred with MariaDB client
connecting to RDS Aurora (MySQL 5.7) database.

intrip committed:
Show a warning when running no migration using SCOPE

When running a migration with `ENV["SCOPE"]` set returns a warning
if no migrations ran.
The message serves as a hint for the end-user to make sure he knows
that the migration have been filtered by `SCOPE`.

andrewculver committed:
Fixing delegated types example.

bjfish committed:
Update test helper to call parallelize according to fork support

samjewell committed:
Update test names to match their behaviour

These tests were names `…creation_failure…` but there's no creation
failing in the tests themselves. Instead creation is succeeding, via
the `create_association` method which is added by the `has_one` relation.

I found these test names very confusing when reading this issue:
https://github.com/rails/rails/issues/13197
And the commit it links to: c6e10b0

The least we can do to make that issue less confusing is to start by
fixing these test names.
[ci skip]

t27duck committed:
Guides: Remove unneeded statement about stylesheet_pack_tag [ci skip]

rafaelfranca committed:
Do not eagerly load Request before initializers

Without those changes the configurations to ActionDispatch::Request
could not be applied in initializers.

rafaelfranca committed:
Change Request#media_type to return nil

When the request don't have a Content-Type header we were returning
an empty string instead of nil like Rack does.

rafaelfranca committed:
`ActionDispatch::Request#content_type` now returned Content-Type header as it is

rafaelfranca committed:
Disable rubocop suggestions

rafaelfranca committed:
Fix typo in the CHANGELOG

rafaelfranca committed:
Add config.action_view.stylesheet_media_default to baseline config

rafaelfranca committed:
Rename the config to make clear it is a boolean

rafaelfranca committed:
Add CHANGELOG entry for #41215

kamipo committed:
Fix CI failure due to `app.config.action_view.delete(:apply_apply_stylesheet_media_default)`

https://buildkite.com/rails/rails/builds/74429#3de35026-a6dc-4f4c-b885-9b59e0c89b96/979-1036

kamipo committed:
Fix deprecation message s/Rails 6.1 will return/Rails 7.0 will return/

`return_only_media_type_on_content_type` will be introduced in Rails 6.2
so the changing of returning Content-Type will happen in a future
version of Rails (probably 7.0).

eileencodes committed:
Revert "Merge pull request #41192 from kamipo/dont_return_pools"

This reverts commit 9d8ff323729f8f8fe09f14f4f45586f2d6340a70, reversing
changes made to 9cde02ef5f17bf4f20aaeb5314de0b52c86496a3.

Need to revert this so I can revert another PR.

eileencodes committed:
Revert "Merge pull request #41046 from eileencodes/dont-check-if-qc-is-enabled"

This reverts commit c97f1f195fc299d9d2ae7954d2b842ec4822382c, reversing
changes made to ac7851eb58d3e821f0f378c1417bc157d7df3be8.

We haven't quite tracked down why yet, but this change caused our API to
not use the query cache. Our API is Sinatra mixed with Rails. We install
the Exectutor in our API so it was installed. However, (some?)
production requests were showing 0 query cache hits.

jhawthorn found that we likely need this check because if we don't the
pools returned will be a different set. He'll send a test later today.

st0012 committed:
Fix typo

Commit:
Update railties/test/generators/scaffold_generator_test.rb

Co-authored-by: Jonathan Hefner <jonathan@hefner.pro>

dhh commented:
I think that's going too far. The reason we didn't use partials the first time around is that the purpose with these scaffolds is soft on-ramp learning. Flash doesn't warrant an extra file for that.

Commit:
Merge branch 'main' into modernize-scaffold

abhaynikam commented on activerecord/CHANGELOG.md:
````diff
@@ -38,7 +38,6 @@
 
     ```ruby
     class User < ApplicationRecord
-      has_many :bookmarks
````
The association was used below to demonstrate usage of strict_loading on a Model. Is the removal intentional or removed by mistake while rebasing the PR?

dhh replied:
mistake, adding back

abhaynikam replied:
Raised a PR for it: https://github.com/rails/rails/pull/41329. Please feel free to close it if you have added the change back 😊

yshmarov commented:
can't wait to use the new scaffold!
